sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala

```diff
@@ -132,7 +135,15 @@ trait GpuPartitioning extends Partitioning {
       }
     }
     withResource(hostPartColumns) { _ =>
-      Cuda.DEFAULT_STREAM.sync()
+      lazy val memCpyNvtxRange = memCopyTime.map(
+          new NvtxWithMetrics("PartitionD2H", NvtxColor.CYAN, _))
```

Comment:
NvtxWithMetrics has an `apply` that already does this for you.

```
withResource(NvtxRange("PartitionD2H", NvtxColor.CYAN, memCopyTime)) { _ =>
...
}
```

Reply:
Fixed.